### PR TITLE
feat(components): [button] add link type

### LINKED ISSUE #7610: [Feature Request] [Enhancement] [button] Add `link` prop
Feature type: **`Enhancement`**

## Details

- Enhanced compone nt name: `button`

## Description

The original website quote is as follows.

> Text button has been upgraded with a new design since 2.2.0 , if you want to use the previous version like button, you might want to check `Link` out.

However, the `Link` component is not exactly the same as the previous text button.

- When using multiple `Link` components, you need to add `margin` manually or use `Space` components to ensure the spacing. When using multiple `Button` components, the spacing is already set and does not need to be handled manually.
- The hover, focus, and active states of the `Link` component are all styled differently from the previous text buttons, including text color, underline, etc.
- The `Link` component renders the `<a>` tag and the `Button` component renders the `<button>` tag.

I think these differences make it impossible to use the `Link` component directly to replace the previous text button in practice. Is it possible to consider adding a `link` property to the `Button` component that corresponds to the previous text button style?

官网原话如下。

> Text button has been upgraded with a new design since 2.2.0 , if you want to use the previous version like button, you might want to check `Link` out.

但是，`Link` 组件和之前的文字按钮并不完全相同。

- 使用多个 `Link` 组件时，需要手动增加 `margin` 或使用 `Space` 组件以保证间距。使用多个 `Button` 组件时，间距已经设置好了，不需要手动处理。
- `Link` 组件的 hover、focus、active 状态下的样式都与之前的文字按钮有所差异，包括文字颜色、下划线等。
- `Link` 组件渲染出 `<a>` 标签，`Button` 组件渲染出 `<button>` 标签。

我认为，这些差异使得实际开发中无法直接使用 `Link` 组件替换之前的文字按钮。是否可以考虑为 `Button` 组件增加 `link` 属性并对应到之前的文字按钮样式？

### Additional comments

EDIT: The original idea of the issue comes from [Ant Design Button](https://ant.design/components/button/) BTW.

### PR DESCRIPTION
**Pull request by chenxch**
Please make sure these boxes are checked before submitting your PR, thank you!

- Closes #7610 

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/theme-chalk/src/button.scss [hunk 1]
```diff
@@ -225,4 +232,30 @@ $button-icon-span-gap: map.merge(
...
+    color: getCssVar('color', 'primary');
+    background: transparent;
+    padding-left: 0;
+    padding-right: 0;
+
+    &:hover,
+    &:focus {
+      color: getCssVar('color', 'primary', 'light-3');
+      border-color: transparent;
+      background-color: transparent;
```
Comment (btea) [inline packages/theme-chalk/src/button.scss:247]: Is `border-color: transparent`、`background-color: transparent` necessary? I see lines 237 and 239 have been added.

File: docs/en-US/component/button.md [hunk 2]
```diff
@@ -109,22 +109,22 @@ button/custom
...
-| auto-insert-space                   | automatically insert a space between two chinese characters     | boolean            |                                                               | —       |
+| Attribute                           | Description                                                     | Type               | Accepted Values                                                      | Default |
+| ----------------------------------- | --------------------------------------------------------------- | ------------------ | -------------------------------------------------------------------- | ------- |
+| size                                | button size                                                     | string             | large / default /small                                               | —       |
+| type                                | button type                                                     | string             | primary / success / warning / danger / info / link / <del>text</del> | —       |
```
Comment (JeremyWuuuuu) [inline docs/en-US/component/button.md:115]:
```suggestion
| type                                | button type                                                     | string             | primary / success / warning / danger / info / link <VersionTag version="2.2.1" /> / <del>text</del> | —       |
```

File: docs/examples/button/text.vue [hunk 3]
```diff
@@ -1,4 +1,13 @@
 <template>
+  <p>Basic button</p>
+  <div class="flex justify-space-between mb-4">
+    <el-button
+      v-for="button in buttons"
+      :key="button.text"
+      :type="button.type"
+      >{{ button.text }}</el-button
+    >
+  </div>
```
Comment (JeremyWuuuuu) [inline docs/examples/button/text.vue:10]: Remove this one and create a new file:

> docs/examples/button/link.vue

For demonstrating link type.

File: packages/theme-chalk/src/button.scss (hunk 1 above)
Comment (JeremyWuuuuu) [reply packages/theme-chalk/src/button.scss:247]: This is just adding back the styles I removed.

**Comment by JeremyWuuuuu** [comment]
Add a breaking change update under  https://github.com/element-plus/element-plus/tree/dev/breakings. 

> breakings/2.2.1/button.yml

File: docs/examples/button/text.vue (hunk 3 above)
Comment (chenxch) [reply docs/examples/button/text.vue]: ok

File: packages/theme-chalk/src/button.scss (hunk 1 above)
Comment (chenxch) [reply packages/theme-chalk/src/button.scss:247]: Yes, just did a restore.

**Pull request closed without merging**